### PR TITLE
Add authentication with tunnistamo

### PR DESCRIPTION
**Pull request by matti-lamppu**
# Hitas Pull Request

# Description

Add the ability to use [tunnistamo](https://github.com/City-of-Helsinki/tunnistamo) as a login method.

~~Note that this has been tested in local environments only. Hitas does not currently have the necessary client setup
in the actual tunnistamo instance, so this will not work yet in dev or test.~~

Also, remember to copy the new values from `.env.template` to your own `.env`!

## Pull request checklist

Check the boxes for each DoD item that has been completed:

- **Testing**
  - [x] Changes have been tested
  - [ ] Automatic tests have been added
- **Database**
  - [ ] Database migrations will work in the DEV & TEST environments
  - [ ] initial.json has been updated to work with migrations
  - [ ] Oracle migration has been updated
- **Documentation**
  - [ ] Tooltips have been added in the frontend for all new fields
  - [ ] OpenAPI definitions have been updated
  - [ ] Test instructions have been written for the customer in the appropriate ticket in Jira
  - [ ] Terminology page in Confluence has been updated

## Test plan

- [ ] Follow the instructions in the `backend/README.md` for setting up tunnistamo.

## Tickets

This pull request resolves all or part of the following ticket(s): HT-457, HT-158

### CODE REVIEW

**Comment by matti-lamppu** [comment]
I only tried this with a backend running locally on `localhost:8085`, and seems like it does not work inside the container yet. I'll make some adjustments tomorrow...

**Review by ranta — CHANGES_REQUESTED** [review]
Please remember to split your commits into smaller, more focused commits. Smaller commits are easier to review and it makes it easier for others to understand the changes that have been made.
For example, this commit added new things, changed some existing code and configs and refactored other code. It should be split into multiple separate commits.

**Comment by matti-lamppu** [comment]
Ok, should be better separated now 👍

File: backend/pyproject.toml [hunk 1]
```diff
@@ -9,6 +9,8 @@ python = "^3.10"
 Django = "^3.2"
 django-environ = "^0.9.0"
 psycopg2 = "^2.9.3"
+drf-oidc-auth = "^3.0.0"
+social-auth-app-django = "5.0.0"
```
Comment (matti-lamppu) [inline backend/pyproject.toml:13]: `social-auth-app-django` version 5.1.0 has some [migration issues](https://github.com/python-social-auth/social-app-django/issues/429) that break our CI, so at the moment this has to be pinned to 5.0.0.

Comment (matti-lamppu) [reply backend/pyproject.toml]: This has been solved with 5.2.0

**Review by ranta — APPROVED** [review]
🔐

**Pull request merged**